File: client-journal/src/app/main/services/page.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from '../../services/appRest.service';
import { Page } from '../model/Page';

@Injectable({
  providedIn: 'root'
})
export class PageService extends AppRestService<Page>{

  constructor(http: HttpClient) {
    super(http, "wordpress");
  }

  getAllPages() {
    return this.getListCached(new HttpParams(), "pages");
  }

}

File: src/main/java/com/jss/osiris/modules/myjss/wordpress/service/PageService.java
package com.jss.osiris.modules.myjss.wordpress.service;

import java.util.List;

import com.jss.osiris.modules.myjss.wordpress.model.Page;

public interface PageService {
        public Page getPage(Integer id);

        public List<Page> getAllPages();

        public Page addOrUpdatePageFromWordpress(Page page);
}
